px-datetime-range-behavior.html

```diff
@@ -191,48 +109,14 @@
    *
    * @return {Boolean} true if chronological order, false otherwise
    */
-  _validateRangeOrder: function() {
+  _validateRangeOrder: function(fromMoment, toMoment) {
     // if the from date is before the to date, everything is ok
-    if(this.fromMoment && this.toMoment) {
-      return this.fromMoment.isBefore(this.toMoment);
+    if(fromMoment && toMoment) {
+      return fromMoment.isBefore(toMoment);
     }else {
       return false;
     }
   },
-  /**
-   * This event is fired differently depending on config:
-   *  - If the px-datetime component has an Apply button, then it is fired when the
-   *   user clicks the Apply button (or presses enter) AND the selected date/time is valid
-   *  - If the px-datetime component does not have an Apply button, then it is fired
-   *   every time a change is made to the date/time AND the selected date/time is valid
-   *
-   * @event px-datetime-range-submitted
-   * @param {object} range - {from, to} the ISO 8601 strings representing the date/time chosen by the user
-   * @param {object} momentObjs - {from, to}: the moment.js objects representing the date/time chosen by the user
-   */
-  _notifyRangeChanged: function(r) {
-    if(r !== undefined) {
-      var newRange = r.value;
-
-      if(!this._preventRangeUpdate) {
-        this._preventRangeUpdate = true;
-
-        //make sure we keep the moment synchronized in case range has been changed externally
-        if(!this.fromMoment || this.fromMoment.toISOString() !== newRange.from) {
-          this.set('fromMoment', this._convertISOtoMoment(newRange.from));
-        }
-        if(!this.toMoment || this.toMoment.toISOString() !== newRange.to) {
-          this.set('toMoment', this._convertISOtoMoment(newRange.to));
-        }
-
-        this.range = newRange;
-
-        //notify the range has changed
-        this.fire('px-datetime-range-submitted', {range: this.range, momentObjs: {from: this.fromMoment, to: this.toMoment}});
```

Comment:
slightly concerned by loosing those 2 events (range and "normal" submitted) which were used to tell the user only about "meaningful" moment changes vs temp changes (typically in a panel and/or with buttons)
Guessing that the idea is now that momentObj will only change when actually needed? Sounds good, might want to pay special attention to that when looking at panel

Reply:
Good point. We aren't sure yet how this works with panel.